source/jormungandr/jormungandr/utils.py

```diff
@@ -524,3 +525,41 @@ def get_house_number(housenumber):
     if len(numbers) > 0:
         hn = numbers[0]
     return int(hn)
+
+
+# The two following functions allow to use flask request context in greenlet
+# The decorator provided by flask (@copy_current_request_context) will generate an assertion error with multiple greenlets
+
+def copy_flask_request_context():
+    """
+    Make a copy of the 'main' flask request conquest to be used with the context manager below
+    :return: a copy of the current flask request context
+    """
+    # Copy flask request context to be used in greenlet
+    top = flask._request_ctx_stack.top
+    if top is None:
+        raise RuntimeError('This function can only be used at local scopes '
+                            'when a request context is on the stack.  For instance within '
+                            'view functions.')
+    return top.copy()
+
+@contextmanager
+def copy_context_in_greenlet_stack(request_context):
```

Comment:
https://stackoverflow.com/questions/48814122/unable-to-copy-current-flask-request-context-to-multiple-gevent-threads/48929494#48929494